src/address/channel.rs

```diff
@@ -497,6 +498,16 @@ impl<A: Actor> PartialEq for AddressSender<A> {
 
 impl<A: Actor> Eq for AddressSender<A> { }
 
+impl<A: Actor> Hash for AddressSender<A> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        let ptr = Arc::into_raw(Arc::clone(&self.inner));
+        ptr.hash(state);
+
+        // Prevent memory leak.
+        unsafe { Arc::from_raw(ptr); }
```

Comment:
If there is no other way than that please describe in comment why we use this unsafe(while it is obvious better to do it)